robot/EDA/resources/EDA.py

```diff
@@ -381,3 +424,187 @@ def get_namespace_prefix(self, name):
+        """ Returns a rendered locator string from the eda_lex_locators
+            dictionary. This can be useful if you want to use an element in
+            a different way than the built in keywords allow.
+        """
+        locator = eda_lex_locators
+        for key in path.split('.'):
+            locator = locator[key]
+        main_loc = locator.format(*args, **kwargs)
+        return main_loc
+
+    def select_navigation_tab(self, tab):
+        """ Selects navigation tab - as passed in to the function """
+        locator_menu = eda_lex_locators["navigation_menu"]
+        element_menu = self.selenium.driver.find_element_by_xpath(locator_menu)
+        locator_tab = eda_lex_locators["navigation_tab"].format(tab)
+
+        self.selenium.wait_until_page_contains_element(
+            locator_menu,
+            error="Navigation menu drop down unavailable"
+        )
+        # javascript is being used here because the usual selenium click is highly unstable for this element on MetaCI
+        self.selenium.driver.execute_script("arguments[0].click()", element_menu)
+        time.sleep(1)
+
+        # Sometimes, single click fails. Hence an additional condition to click on it again
+        if not self.check_if_element_exists(locator_tab):
+            self.selenium.driver.execute_script("arguments[0].click()", element_menu)
+            time.sleep(1)
+
+        self.selenium.capture_page_screenshot()
+        self.selenium.wait_until_page_contains_element(
+            locator_tab,
+            error=tab + " item not found as an available option in: " + locator_tab
+        )
+        self.selenium.click_element(locator_tab)
+
+
+
+    def select_tab(self, tab):
+        """ Selects Contacts tab - as passed in to the function """
+        locator_menu = eda_lex_locators["tab_menu"].format(tab)
+        element_menu = self.selenium.driver.find_element_by_xpath(locator_menu)
+        locator_tab = eda_lex_locators["tab_tab"].format(tab)
+        self.selenium.wait_until_page_contains_element(
+            locator_menu,
+            error="Contact tab unavailable"
+        )
+        # javascript is being used here because the usual selenium click is highly unstable for this element on MetaCI
+        self.selenium.driver.execute_script("arguments[0].click()", element_menu)
+        time.sleep(1)
+
+        # Sometimes, single click fails. Hence an additional condition to click on it again
+        if not self.check_if_element_exists(locator_tab):
+            self.selenium.driver.execute_script("arguments[0].click()", element_menu)
+            time.sleep(1)
+
+
+    def select_value_from_listbox(self, title, value):
+        """ Selects value from a listbox/dropdown identified by title """
+        locator_title = eda_lex_locators["listbox"]["title"].format(title)
+        locator_value = eda_lex_locators["listbox"]["value"].format(value)
+        self.selenium.click_element(locator_title)
+        time.sleep(1)
```

Comment:
Instead of putting in a hard-coded sleep, wouldn't it be better to wait for the item to be visible?

Reply:
This will have to be looked at and fixed in the EDA rewrite, since it was written by someone else, and is outside the scope of Preferred Phone script.  It would be nice to fix all the bugs in this code written by someone else, but this particular work item needs to limit the scope to the new feature work for Preferred Phone.

Reply:
I personally don't think that's the right thing to do. If you're adding code to the project, you should be responsible for adding good code. Ultimately it's your decision, however.

Reply:
I removed this function